cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -553,10 +539,9 @@ bool reader::impl::has_next()
                {} /*row_group_indices, empty means read all row groups*/,
                std::nullopt /*filter*/);
 
-  size_t const num_input_passes = std::max(
-    int64_t{0}, static_cast<int64_t>(_file_itm_data.input_pass_row_group_offsets.size()) - 1);
-  return (_pass_itm_data->current_output_chunk < _pass_itm_data->output_chunk_read_info.size()) ||
-         (_current_input_pass < num_input_passes);
+  // current_input_pass will only be incremented to be == num_passes after
+  // the last chunk in the last subpass in the last pass has been returned
+  return has_more_work();
```

Comment:
Nit: I feel has "more work" is a bit vague. Does that mean has more data to process?

Reply:
Right. Work can mean a number of things:  actual IO, decompression, decoding, etc. It does generally mean "more data to process and return"